Editor/Scripts/DCCIntegration/AutoServerCreation/AutoSetup.cs

```diff
@@ -0,0 +1,95 @@
+using System;
+using System.Threading;
+using UnityEditor;
+using UnityEngine;
+using Object = UnityEngine.Object;
+
+namespace Unity.MeshSync.Editor {
+
+[InitializeOnLoad]
+public static class AutoSetup {
+
+    private static Server m_server;
+    
+    static AutoSetup() {
+        // Create server
+        var settings = ServerSettings.defaultValue;
+        
+        settings.port = AutoSetupSettings.instance.Port;
+        
+        if (!Server.Start(ref settings, out m_server)) {
+            Debug.LogError("[MeshSync] Could not start Server");
+            return;
+        }
+        
+        //Register calls
+        EditorApplication.update -= UpdateCall;
+        EditorApplication.update += UpdateCall;
+    }
+
+    private static void UpdateCall() {
+       if (m_server.numMessages == 0)
+           return;
+        
+       m_server.ProcessMessages(Handler);
+    }
+
+    private static void Handler(NetworkMessageType type, IntPtr data) {
+        switch (type) {
+            case NetworkMessageType.EditorCommand:
+                OnRecvEditorCommand((EditorCommandMessage)data);
+                break;
+            default:
+                Debug.LogErrorFormat("[MeshSync] Unhandled message type {0}", type);
+                break;
+        }
+    }
+
+    private static void OnRecvEditorCommand(EditorCommandMessage message) {
+        var type = message.commandType;
+        
+        switch (type) {
+            case EditorCommandMessage.CommandType.AddServerToScene:
+                HandleAddServerToScene();
+                break;
+            case EditorCommandMessage.CommandType.GetProjectPath:
+                HandleGetProjectPath();
+                break;
+            default:
+                Debug.LogErrorFormat("[MeshSync] Unhandled command type {0}", type);
+                break;
+        }
+    }
+
+    private static void HandleAddServerToScene() {
+        if (AddServerToScene()) {
+            m_server.NotifyEditorCommand("ok");
+        }
+        else {
+            m_server.NotifyEditorCommand("Could not start server");
+            Debug.LogErrorFormat("[MeshSync] Could not add server to scene");
+        }
+    }
+
+    private static void HandleGetProjectPath() {
+        var path = GetProjectPath();
+        m_server.NotifyEditorCommand(path);
+    }
+
+    private static string GetProjectPath() {
```

Comment:
I think this will break the code: I will always return null as Application.dataPath does not start with "Assets/"
```
    /// <summary>
    /// If the path starts with "Assets/" then returns the relative path under Assets.
    /// Otherwise, returns null.
    /// </summary>
```

Reply:
Ah, sorry, I think I misunderstood the intent of this function, despite its name.
It should be `FilmInternalUtilities.AssetEditorUtility.GetApplicationRootPath()`

Reply:
Oh nice, will replace it

Reply:
Right...
Can you please copy `FilmInternalUtilities.Editor.AssetEditorUtility` function as it is to meshsync side and add `[TODO-sin]` comment ?
I'll take care of removing this duplicate code later

Reply:
done